internal/cmd/controller/reconciler/gitrepo_controller.go

```diff
@@ -244,19 +266,9 @@ func (r *GitRepoReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 				return []ctrl.Request{}
 			}),
+			builder.WithPredicates(bundleStatusChangedPredicate()),
```

Comment:
Nice, this means that we could probably skip the check on the object being a bundle within `bundleStatusChangedPredicate`.